include/circt/Dialect/OM/OMAttributes.td

```diff
@@ -81,4 +81,26 @@ def OMListAttr : AttrDef<OMDialect, "List", [TypedAttrInterface]> {
   }];
 }
 
+def MapAttr : AttrDef<OMDialect, "Map", [TypedAttrInterface]> {
+  let summary = "An attribute that represents a string map";
+
+  let mnemonic = "map";
+
+  let parameters = (ins
+    "mlir::Type": $valueType,
+    "mlir::DictionaryAttr":$elements
+  );
```

Comment:
The `valueType` is the type of the values in the map, yes? Is this going to work for OMIR stuff where I think the maps are heterogeneous? I.e., I thought the values could be anything.

Reply:
> The valueType is the type of the values in the map, yes?

Yes.

> OMIR stuff where I think the maps are heterogeneous?
 Oh, interesting.  If that's the case introducing`!om.any` might be better option. It's currently align with `firrtl.map` type so `firrtl.map` also needs to be fixed as well.

Reply:
I'm going to just merge the PR to unblock @nandor's use case since we rather need to modify the type system itself 👍